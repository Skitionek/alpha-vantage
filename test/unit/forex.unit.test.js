
import AlphaVantageMock from "../mocks/alphaVantageMock";
import { matchesSnapshot } from "../jest.extensions";

jest.setTimeout(30000);

let alpha;
beforeAll(() => {
	alpha = new AlphaVantageMock();
});

it(`daily data works`, () => {
	return alpha.forex.daily('btc', 'usd')
		.then(matchesSnapshot);
});

it(`exchangeRates data works`, () => {
	return alpha.forex.exchangeRates('btc', 'usd')
		.then(matchesSnapshot);
});

describe(`exchangeTimeSeries data works`, () => {
	it('5min', () =>
		alpha.forex.exchangeTimeSeries({
			from_symbol:'btc',
			to_symbol: 'usd',
			interval: '5min'
		})
			.then(matchesSnapshot)
	);
});

it(`intraday data works`, () => {
	return alpha.forex.intraday('btc', 'usd')
		.then(matchesSnapshot);
});

it(`monthly data works`, () => {
	return alpha.forex.monthly('btc', 'usd')
		.then(matchesSnapshot);
});

it(`weekly data works`, () => {
	return alpha.forex.weekly('btc', 'usd')
		.then(matchesSnapshot);
});