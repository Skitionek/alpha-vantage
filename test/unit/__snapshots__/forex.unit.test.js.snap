// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`daily data works 1`] = `
Object {
  "data": Array [
    Object {
      "close": "Float",
      "date": "Timestamp/Date",
      "high": "Float",
      "low": "Float",
      "open": "Float",
    },
  ],
  "from_symbol": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "output_size": "String",
  "time_zone": "String",
  "to_symbol": "String",
}
`;

exports[`exchangeRates data works 1`] = `
Object {
  "ask_price": "Float",
  "bid_price": "Float",
  "exchange_rate": "Float",
  "from_currency_code": "String",
  "from_currency_name": "String",
  "last_refreshed": "Timestamp/Date",
  "time_zone": "String",
  "to_currency_code": "String",
  "to_currency_name": "String",
}
`;

exports[`exchangeTimeSeries data works 5min 1`] = `
Object {
  "data": Array [
    Object {
      "close": "Float",
      "date": "Timestamp/Date",
      "high": "Float",
      "low": "Float",
      "open": "Float",
    },
  ],
  "from_symbol": "String",
  "information": "String",
  "interval": "Interval",
  "last_refreshed": "Timestamp/Date",
  "output_size": "String",
  "time_zone": "String",
  "to_symbol": "String",
}
`;

exports[`intraday data works 1`] = `
Object {
  "data": Array [
    Object {
      "close": "Float",
      "date": "Timestamp/Date",
      "high": "Float",
      "low": "Float",
      "open": "Float",
    },
  ],
  "from_symbol": "String",
  "information": "String",
  "interval": "Interval",
  "last_refreshed": "Timestamp/Date",
  "output_size": "String",
  "time_zone": "String",
  "to_symbol": "String",
}
`;

exports[`monthly data works 1`] = `
Object {
  "data": Array [
    Object {
      "close": "Float",
      "date": "Timestamp/Date",
      "high": "Float",
      "low": "Float",
      "open": "Float",
    },
  ],
  "from_symbol": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "time_zone": "String",
  "to_symbol": "String",
}
`;

exports[`weekly data works 1`] = `
Object {
  "data": Array [
    Object {
      "close": "Float",
      "date": "Timestamp/Date",
      "high": "Float",
      "low": "Float",
      "open": "Float",
    },
  ],
  "from_symbol": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "time_zone": "String",
  "to_symbol": "String",
}
`;
